lib/rubocop/cop/autocorrect_logic.rb

```diff
@@ -0,0 +1,22 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    # This module encapsulates the logic for autocorrect behaviour for a cop
+    module AutocorrectLogic
+      def autocorrect?
```

Comment:
I'm not sure extracting this into a module is a good idea. After all, this functionality doesn't make sense outside anything but a cop.